core/src/main/java/org/elasticsearch/index/translog/Translog.java

```diff
@@ -1522,6 +1576,10 @@ public void trimUnreferencedReaders() throws IOException {
                 // we're shutdown potentially on some tragic event, don't delete anything
                 return;
             }
+            if (readers.isEmpty()) {
```

Comment:
yeah, I can do that and thought about it. The reason I went with this is that to do something like `shouldTrimTranslog` where I can add this shortcut, I will also have to go and ask the retention policy for the current trimming situation and add tests etc. Since this method is not called on every write, but I rather on every roll / flush/ background check every 10m.. I felt this was simpler. Let me know what you think.